src/lexer.ts

```diff
@@ -255,6 +273,16 @@ export class Lexer {
       };
     }
 
+    // What if we have an assignment operator?
+    if (OPERATORS.ASSIGNMENT === currentIdentifier) {
+      return {
```

Comment:
TODO: we will need to come here and follow up with `peaking` to see if the next char is `=` and to treat it as an `==` operator, similar to how we are doing `--` for comments.